### PR TITLE
Adding new api keys and oauth pages

### LINKED ISSUE #2266: Implement header JS part in Api and OAuth pages
Only header part please.
Branch [new-api-keys](https://github.com/CartoDB/cartodb/tree/new-api-keys), pull request created -> https://github.com/CartoDB/cartodb/pull/2142#issuecomment-74679491

### PR DESCRIPTION
**Pull request by xavijam**
New API keys + OAuth credentials pages.
Please @kartones, @juanignaciosl or @rafatower, could you check those first changes in Ruby are ok?

fixes #2266

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1470/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1471/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1472/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1473/))

File: app/controllers/admin/client_applications_controller.rb [hunk 1]
```diff
@@ -34,4 +44,18 @@ def regenerate_api_key
     redirect_to api_key_credentials_path(user_domain: params[:user_domain], type: 'api_key'), :flash => {:success => "Your API key has been regenerated successfully"}
   end
 
+  def regenerate_oauth
+    @client_application = current_user.client_application
+    return if request.get?
+    current_user.reset_client_application!
+    
+    new_dashboard = current_user.has_feature_flag?('new_dashboard')
+    if new_dashboard
```
Comment (juanignaciosl) [inline app/controllers/admin/client_applications_controller.rb]: Do you really want to redirect to one page or another depending on `new_dashboard` flag?

File: app/views/admin/client_applications/new_oauth.html.erb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        </div>
+
+        <span class="Form-separator"></span>
+
+        <div class="Form-row">
+          <div class="Form-rowLabel">
+            <label class="Form-label">Request token URL</label>
+          </div>
+          <div class="Form-rowData">
+            <input type="text" class="Form-input Form-input--long" value="https://<%= current_user.username %>.<%= Cartodb.config[:account_host] %>/oauth/request_token" readonly />
```
Comment (Kartones) [inline app/views/admin/client_applications/new_oauth.html.erb]: Sorry but no handcrafted urls if can use Rails "url builders". Use instead `request_token_url` (here), 'access_token_url' (just below at line 86), etc. (See `routes.rb` file, around line 20, the `as: xxxx` part. 

I can explain in detail how this works but REST API urls must always be sent from rails (especially having the URL rewrite project halfway done).

File: config/routes.rb [hunk 3]
```diff
@@ -248,7 +248,8 @@
 
     match  '(/u/:user_domain)/your_apps'                    => 'client_applications#api_key',            as: :api_key_credentials
     post   '(/u/:user_domain)/your_apps/api_key/regenerate' => 'client_applications#regenerate_api_key', as: :regenerate_api_key
-    delete '(/u/:user_domain)/your_apps/oauth'              => 'client_applications#oauth',              as: :oauth_credentials
```
Comment (Kartones) [inline config/routes.rb]: I guess this was a bug before, no?

File: app/views/admin/client_applications/new_oauth.html.erb (hunk 2 above)
Comment (Kartones) [inline app/views/admin/client_applications/new_oauth.html.erb]: Above at `new_api_key.html.erb` is done correctly so is just doing the same here.

**Comment by rafatower** [comment]
Nothing to add to what's already said

File: app/controllers/admin/client_applications_controller.rb (hunk 1 above)
Comment (xavijam) [inline app/controllers/admin/client_applications_controller.rb]: I have to preserve the old api keys behaviour (preventing legacy problems), so yes, :(

File: app/views/admin/client_applications/new_oauth.html.erb (hunk 2 above)
Comment (xavijam) [inline app/views/admin/client_applications/new_oauth.html.erb]: Roger :+1:

File: config/routes.rb (hunk 3 above)
Comment (xavijam) [inline config/routes.rb]: It was strange, I feel we joined api keys and oauth pages in the last redesign and we left routes a bit weird.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1480/))

**Comment by xavijam** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1483/))

**Comment by javisantana** [comment]
please update news if needed and merge

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1705/))

**Comment by viddo** [comment]
@xavijam please review last changes, basically extracted context-specific logic to a view model so the header can be reused in the two new contexts (dashboard and app/OAuth keys pages)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1706/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1708/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1713/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1714/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1717/))

**Comment by xavijam** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1727/))

**Pull request merged**